salt/serializers/msgpack.py

```diff
@@ -63,7 +57,13 @@ def _serialize(obj, **options):
     def _deserialize(stream_or_string, **options):
         try:
             options.setdefault("use_list", True)
-            options.setdefault("encoding", "utf-8")
+
+            # msgpack deprecated `encoding` starting with version 0.5.2
+            if salt.utils.msgpack.version >= (0, 5, 2):
+                # Under Py2 we still want raw to be set to True
+                options.setdefault("raw", six.PY2)
```

Comment:
we no longer support python2, so we do not need to add support for this. Also we should be handling this  in salt/utils/msgpack.py instead so it handles all cases of msgpack using encoding.

As you can see we are already doing this for versions 1.0 https://github.com/saltstack/salt/blob/master/salt/utils/msgpack.py#L72 and here  https://github.com/saltstack/salt/blob/master/salt/utils/msgpack.py#L86